erpnext/patches/v15_0/update_total_number_of_booked_depreciations.py

```diff
@@ -0,0 +1,26 @@
+import frappe
+
+from erpnext.assets.doctype.asset_depreciation_schedule.asset_depreciation_schedule import (
+	get_depr_schedule,
+)
+
+
+def execute():
+	if frappe.db.has_column("Asset Finance Book", "total_number_of_booked_depreciations"):
+		assets = frappe.get_all(
+			"Asset", filters={"docstatus": 1}, fields=["name", "opening_number_of_booked_depreciations"]
+		)
+
+		for asset in assets:
+			asset_doc = frappe.get_doc("Asset", asset.name)
+
+			for fb_row in asset_doc.get("finance_books"):
+				depr_schedule = get_depr_schedule(asset.name, "Active", fb_row.finance_book)
+				total_number_of_booked_depreciations = asset.opening_number_of_booked_depreciations or 0
+
+				for je in depr_schedule:
+					if je.journal_entry:
+						total_number_of_booked_depreciations += 1
+				fb_row.db_set("total_number_of_booked_depreciations", total_number_of_booked_depreciations)
+			asset_doc.save
+		frappe.db.commit()
```

Comment:
explicit commit is not required, framework auto commits after running the patch.